source/git-api.js

```diff
@@ -692,23 +692,33 @@ exports.registerApi = (env) => {
     });
     app.post(`${exports.pathPrefix}/testing/createfile`, ensureAuthenticated, (req, res) => {
       const content = req.body.content ? req.body.content : (`test content\n${Math.random()}\n`);
-      fs.writeFile(req.body.file, content, () => res.json({}));
+      fs.writeFileAsync(req.body.file, content)
+        .then(() => res.json({}))
+        .then(emitWorkingTreeChanged.bind(null, req.body.path));
```

Comment:
To make the tests more reliable trigger working tree changes manually.

Reply:
I don't particularly like the `bind`. Would this work too?

```suggestion
        .then(() => emitWorkingTreeChanged(req.body.path));
```

Reply:
Yes, it would but I just want to keep the existing codestyle for now.
In the future we will replace all of this with async await anyway.